features/storage/kvstore/tdbstore/TDBStore.cpp

```diff
@@ -938,6 +938,11 @@ int TDBStore::build_ram_table()
     offset = _master_record_offset;
 
     while (offset < _free_space_offset) {
+        if ((offset + sizeof(record_header_t)) >= _free_space_offset) {
```

Comment:
Why not to embed this into the `while()` condition?

```
while (offset + sizeof(record_header_t) < _free_space_offset) {
```

Reply:
You must set a return code so that the incorrect data is overwritten (corrected) in the TDBStore :: init () function

Reply:
Not really, `TDBStore::init()` calls `build_ram_table()` only once and the logic is to find all valid keys.
https://github.com/ARMmbed/mbed-os/blob/master/features/storage/kvstore/tdbstore/TDBStore.cpp#L1121..L1125

When `build_ram_table()` reaches to a point where data is either corrupted, or erased, it does not matter, it stops. And this new location is considered as end of valid keys, known as `_free_space_offset`.

So whether the `build_ram_table()` returns `SUCCESS` or `CORRUPTED_DATA_FOUND` does not matter. They both lead to same result.

Reply:
`They both lead to same result` - Ok. Previously, the difference was